core/include/detray/propagator/navigation_policies.hpp

```diff
@@ -0,0 +1,90 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// detray definitions
+#include "detray/definitions/qualifiers.hpp"
+#include "detray/propagator/base_actor.hpp"
+
+// system includes
+#include <climits>
+
+namespace detray {
+
+/// Struct that represents the most conservative navigation policy: alway re-
+/// initialize the current volume
+struct always_init : actor {
+
+    struct state_type {};
+
+    /// Sets the navigation trust level to 'no trust'
+    ///
+    /// @param pol_state not used
+    /// @param propagation state of the propagation
+    template <typename propagator_state_t>
+    DETRAY_HOST_DEVICE inline void operator()(
+        const state_type & /*pol_state*/,
+        propagator_state_t &propagation) const {
+        propagation._navigation.set_no_trust();
+    }
+};
+
+/// During guided navigation only the next surface should be re-evaluated. This
+/// maps to the 'high trust' level in the navigator
+struct guided_navigation : actor {
+
+    struct state_type {};
+
+    /// Sets the navigation trust level to 'no trust'
+    ///
+    /// @param pol_state not used
+    /// @param propagation state of the propagation
+    template <typename propagator_state_t>
+    DETRAY_HOST_DEVICE inline void operator()(
+        const state_type & /*pol_state*/,
+        propagator_state_t &propagation) const {
+        propagation._navigation.set_high_trust();
```

Comment:
Indeed, we need to do that. The navigator handles this automatically (it calls ```check_volume_switch``` in every update). This policy also really only makes sense in guided navigation, where we only have the inital volume local navigation (which must be a brute force one instead of a grid).

Reply:
OK